### PR TITLE
auditd: fix service ordering

### PR DESCRIPTION
**Pull request by r-vdp**
## Description of changes

This is needed so that:
  - users have been created (when using systemd-sysusers or userborn)
  - /run and /var/run exist

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/security/auditd.nix [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
-  config = mkIf config.security.auditd.enable {
+  config = lib.mkIf config.security.auditd.enable {
     boot.kernelParams = [ "audit=1" ];
 
     environment.systemPackages = [ pkgs.audit ];
 
     systemd.services.auditd = {
       description = "Linux Audit daemon";
       wantedBy = [ "basic.target" ];
+      wants = [
```
Comment (nikstur) [inline nixos/modules/security/auditd.nix:19]: We shouldn't need wants here. Setting only `after` should be enough.

**Review by nikstur — APPROVED** [review]

File: nixos/modules/security/auditd.nix (hunk 1 above)
Comment (r-vdp) [reply nixos/modules/security/auditd.nix:19]: Even with `DefaultDependencies = no` ? How do we make sure then that this service is not started before the tmpfiles one got added to the transaction? Or does the whole transaction always get created before anything starts executing?

Comment (nikstur) [reply nixos/modules/security/auditd.nix:19]: tmpfiles is pulled in by sysinit.target. By the time basic.target is started, tmpfiles should've already been pulled in. Then it should only be an ordering issue and thus only require `after`. Also compare the upstream service config: https://github.com/linux-audit/audit-userspace/blob/master/init.d/auditd.service.in#L19

In general it seems that auditd should probably be:

```nix
wantedBy = [ "sysinit.target" ];
before = [ "sysinit.target" ];
after = [ "systemd-tmpfiles-setup.service" ];
```

Comment (r-vdp) [reply nixos/modules/security/auditd.nix]: Done. There's a couple of other things that we might want to take from the upstream unit then, but I'll do that as a separate PR.

**Pull request merged**